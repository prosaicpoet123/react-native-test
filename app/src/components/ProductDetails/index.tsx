import React, { FC } from 'react'
import { RootStackParamList } from '../../@Types/navigation'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import ProductCard from '../ProductCard'

type IProps = NativeStackScreenProps<RootStackParamList, 'Product Details'>

const ProductDetails: FC<IProps> = ({ route }) => {
  const { product } = route.params

  return <ProductCard product={product} resolution={370} isDetails />
}

export default ProductDetails
