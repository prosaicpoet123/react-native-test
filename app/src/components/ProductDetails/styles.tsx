import { StyleSheet } from 'react-native'
import { colours } from '../../config/themes'

const styles = StyleSheet.create({
  container: {
    backgroundColor: colours.white,
  },
})

export default styles
