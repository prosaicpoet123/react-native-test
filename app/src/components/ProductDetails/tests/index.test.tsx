import React from 'react'
import renderer from 'react-test-renderer'
import ProductDetails from '../'

describe('<ProductDetails />', () => {
  it('renders correctly', () => {
    const route = {
      name: 'Product Details',
      key: 'Product Details',
      params: {
        product: {
          id: '000001',
          name: 'Product Name',
          images: [],
        },
      },
    }
    const tree = renderer.create(<ProductDetails route={route} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
