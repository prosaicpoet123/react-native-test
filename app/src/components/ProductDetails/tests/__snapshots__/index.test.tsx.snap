// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetails /> renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CECECE",
              "borderWidth": 1,
              "minHeight": 300,
              "padding": 4,
            },
            Object {
              "borderWidth": 0,
            },
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://images.hollandandbarrettimages.co.uk/productimages/HB/370/000001_A.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "height": 180,
                    "width": "100%",
                  },
                  Object {
                    "height": 370,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginVertical": 8,
              }
            }
          >
            <RCTScrollView
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={true}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View />
            </RCTScrollView>
          </View>
        </View>
        <View
          style={
            Object {
              "padding": 4,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "marginBottom": 8,
                },
                Object {
                  "fontWeight": "bold",
                },
                Object {
                  "fontSize": 22,
                },
              ]
            }
          >
            Product Name
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
